server/jetstream_cluster.go

```diff
@@ -1383,6 +1386,11 @@ func (js *jetStream) applyStreamEntries(mset *Stream, ce *CommittedEntry, isReco
 				if err != nil {
 					panic(err.Error())
 				}
+				// Ignore if we are recovering and we have already processed.
+				if isRecovering && mset.State().LastSeq > sp.LastSeq {
```

Comment:
I think this is right, however, say that purge was done at a time that stream had messages 1..100, and now stream is at seq 150. Then on replay there would be no purge. But it is possible that this node did not play the logs in the first place so it would add messages 1..150. I thought you had a purge(seq) that would have purged messages up to that point. I think you need that.
Even with a purge(seq) you should still verify that the mset's state *FirstSeq* is above the sp.LastSeq to completely ignore.

Reply:
In other words, you still should ensure that messages up to sp.LastSeq are no longer in the mset.

Reply:
ok will take a different look at it..

Reply:
ok will compact regardless which will be a no-op if was successful first time processing.

Reply:
ok I think I have it correct now.